File: backend/models/property.py
from flask_sqlalchemy import SQLAlchemy
from datetime import datetime, timezone

db = SQLAlchemy()

class Property(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    title = db.Column(db.String(200), nullable=False)
    description = db.Column(db.Text, nullable=False)
    price = db.Column(db.Float, nullable=False)
    bedrooms = db.Column(db.Integer)
    bathrooms = db.Column(db.Float)
    square_feet = db.Column(db.Float)
    address = db.Column(db.String(300), nullable=False)
    city = db.Column(db.String(100), nullable=False)
    state = db.Column(db.String(50), nullable=False)
    zip_code = db.Column(db.String(10), nullable=False)
    property_type = db.Column(db.String(50))  # house, apartment, condo, etc.
    listing_type = db.Column(db.String(20))   # sale or rent
    created_at = db.Column(db.DateTime, default=lambda: datetime.now(timezone.utc))
    updated_at = db.Column(db.DateTime, default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc))

    def to_dict(self):
        return {
            'id': self.id,
            'title': self.title,
            'description': self.description,
            'price': self.price,
            'bedrooms': self.bedrooms,
            'bathrooms': self.bathrooms,
            'square_feet': self.square_feet,
            'address': self.address,
            'city': self.city,
            'state': self.state,
            'zip_code': self.zip_code,
            'property_type': self.property_type,
            'listing_type': self.listing_type,
            'created_at': self.created_at.isoformat(),
            'updated_at': self.updated_at.isoformat()
        }

File: frontend/src/types/property.ts
/**
 * Represents a property
 * @interface Property
 */
export interface Property {
  /**
   * Unique identifier
   */
  id: number;
  /**
   * Property title/name
   */
  title: string;
  /**
   * Detailed description
   */
  description: string;
  /**
   * Array of image URLs
   */
  images: string[];
  /**
   * Price in USD
   */
  price: number;
  /**
   * Number of bedrooms
   */
  bedrooms: number;
  /**
   * Number of bathrooms
   */
  bathrooms: number;
  /**
   * Total square footage
   */
  square_feet: number;
  /**
   * Street address
   */
  address: string;
  /**
   * City name
   */
  city: string;
  /**
   * State or province
   */
  state: string;
  /**
   * Postal/ZIP code
   */
  zip_code: string;
  /**
   * Type of property
   */
  property_type: string;
  /**
   * Listing type
   */
  listing_type: string;
  /**
   * ISO date string of when listing was created
   */
  created_at: string;
  /**
   * ISO date string of last update
   */
  updated_at: string;
  /**
   * Date the property was listed
   */
  listDate: string;
  /**
   * Virtual tour URL if available
   */
  virtualTourUrl?: string;
  /**
   * Latitude coordinate
   */
  latitude: number;
  /**
   * Longitude coordinate
   */
  longitude: number;
}